src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -60,5 +65,84 @@ public ConnectionInfo(ISqlConnectionFactory factory, string ownerUri, Connection
         /// Returns true is the db connection is to a SQL db
         /// </summary>
         public bool IsAzure { get; set; }
+
+        /// <summary>
+        /// Try to get the DbConnection associated with the given connection type string. 
+        /// </summary>
+        public bool TryGetConnection(string connectionType, out DbConnection connection)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+
+            return ConnectionTypeToConnectionMap.TryGetValue(connectionType, out connection);
+        }
+
+        /// <summary>
+        /// Get a List of all DbConnection instances held by this ConnectionInfo
+        /// </summary>
+        public List<DbConnection> GetAllConnections()
+        {
+            return ConnectionTypeToConnectionMap.Values.ToList();
+        }
+
+        /// <summary>
+        /// Get a list of all connection type strings held by this ConnectionInfo
+        /// </summary>
+        /// <returns></returns>
+        public List<string> GetAllConnectionTypes()
+        {
+            return ConnectionTypeToConnectionMap.Keys.ToList();
+        }
+
+        /// <summary>
+        /// Gets the count of DbConnectioninstances held by this ConnectionInfo 
+        /// </summary>
+        public int CountConnections()
+        {
+            return ConnectionTypeToConnectionMap.Count;
+        }
+
+        /// <summary>
+        /// Adds a DbConnection to this object and associates it with the given 
+        /// connection type string. If a connection already exists with an identical 
+        /// connection type string, it is overwritten. Ignores calls where connectionType = null
+        /// </summary>
+        public void AddConnection(string connectionType, DbConnection connection)
+        {
+            if (connectionType != null)
+            {
+                ConnectionTypeToConnectionMap.Add(connectionType, connection);
+            }
+        }
+
+        /// <summary>
+        /// If string connectionType is not null, removes the single DbConnection
+        /// instance associated with string connectionType. If string connectionType
+        /// is null, removes all DbConnection instances. 
+        /// </summary>
+        /// <param name="connectionType"></param>
+        /// <returns>true if there are no more DbConnection instances held
+        /// by this object after trying to remove the requested connection(s),  
+        /// false otherwise</returns>
+        public bool RemoveConnection(string connectionType)
+        {
+            // Remove a single DbConnection
+            if (connectionType != null)
+            {
+                ConnectionTypeToConnectionMap.Remove(connectionType);
+            }
+            // Remove all DbConnections 
+            else
```

Comment:
not sure have connectionType == null is the best way for the user to specify remove all connections.  May connectionType == null should throw an exception and we have a different overload for Remove All